includes/wp-cli/class-reset.php

```diff
@@ -0,0 +1,373 @@
+	 *
+	 * ## OPTIONS
+	 *
+	 * [--yes]
+	 * : Skip prompt to confirm deletion.
+	 *
+	 * [--all]
+	 * : Delete all published posts, pages, custom posts and media, not just posts and media associated with ACM models.
+	 *
+	 * ## EXAMPLES
+	 *
+	 *     wp acm reset
+	 *     wp acm reset --yes
+	 *     wp acm reset --all
+	 *     wp acm reset --yes --all
+	 *
+	 * @param array $args Options passed to the command, keyed by integer.
+	 * @param array $assoc_args Options keyed by string.
+	 */
+	public function __invoke( $args, $assoc_args ) {
+		\WP_CLI::confirm( 'Delete ACM data, including posts, media, taxonomies and models?', $assoc_args );
+
+		$delete_all = $assoc_args['all'] ?? false;
+
+		\WP_CLI::log( 'Deleting taxonomy terms.' );
+		$this->delete_taxonomy_terms();
+
+		\WP_CLI::log( 'Deleting ACM taxonomies.' );
+		$this->delete_taxonomies();
+
+		\WP_CLI::log( 'Deleting relationship data.' );
+		$this->delete_relationships();
+
+		\WP_CLI::log( 'Deleting posts.' );
+		$this->delete_posts( $delete_all );
+
+		\WP_CLI::log( 'Deleting media.' );
+		$this->delete_media( $delete_all );
+
+		\WP_CLI::log( 'Deleting models.' );
+		$this->delete_models();
+
+		$this->log_stats();
+		\WP_CLI::success( 'ACM reset complete.' );
+	}
+
+	/**
+	 * Deletes taxonomy terms
+	 */
+	public function delete_taxonomy_terms() {
+		foreach ( $this->taxonomies as $taxonomy ) {
+			$terms = get_terms(
+				[
+					'taxonomy'   => $taxonomy['slug'],
+					'hide_empty' => false,
+				]
+			);
+			foreach ( $terms as $term ) {
+				if ( (bool) wp_delete_term( $term->term_id, $taxonomy['slug'] ) ) {
+					$this->stats['taxonomy_terms']++;
+				}
+			}
+		}
+	}
+
+	/**
+	 * Deletes ACM taxonomies.
+	 */
+	public function delete_taxonomies() {
+		if ( delete_option( self::ACM_TAXONOMY_OPTION ) ) {
+			$this->stats['taxonomies'] = count( (array) $this->taxonomies );
+			$this->taxonomies          = [];
+		}
+	}
+
+	/**
+	 * Deletes all ACM relationship data.
+	 */
+	public function delete_relationships() {
+		global $wpdb;
+
+		$table        = ContentConnect::instance()->get_table( 'p2p' );
+		$post_to_post = esc_sql( $table->get_table_name() );
+
+		$deleted_rows = $wpdb->query("DELETE FROM `{$post_to_post}`;"); // phpcs:ignore -- table name escaped earlier.
+
+		if ( is_int( $deleted_rows ) ) {
+			$this->stats['relationships'] = $deleted_rows;
+		}
+	}
+
+	/**
+	 * Deletes published posts.
+	 *
+	 * @param bool $delete_all Pass true to also delete post types unrelated to
+	 *                         ACM, such as core posts and pages.
+	 */
+	public function delete_posts( $delete_all = false ) {
+		if ( $delete_all ) {
+			$posts = new \WP_Query(
+				[
+					'post_type'      => 'any',
+					'posts_per_page' => -1,
+				]
+			);
+
+			foreach ( $posts->posts as $post ) {
+				if ( (bool) wp_delete_post( $post->ID, true ) ) {
+					$this->stats['posts']++;
+				}
+			}
```

Comment:
To avoid the second loop, like you did in `delete_media()`.
```suggestion
			}
			
			return;
```

Reply:
Do the `test_posts_are_reset` and `test_core_posts_are_reset` tests pass for you with this line? For me the tests were failing with this return statement. I think the second loop still needs to run to catch private ACM posts.

Reply:
Yeah they fail for me too. I see why now that I've pulled it down and stepped through it.